frontend/src/components/register/RegisterForm.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Field, reduxForm } from 'redux-form';
+
+const required = value => (value ? undefined : 'Required')
+const maxLength = max => value =>
+  value && value.length > max ? `Must be ${max} characters or less` : undefined
+const maxLength15 = maxLength(15)
+export const minLength = min => value =>
+  value && value.length < min ? `Must be ${min} characters or more` : undefined
+export const minLength2 = minLength(2)
+const number = value =>
+  value && isNaN(Number(value)) ? 'Must be a number' : undefined
+const minValue = min => value =>
+  value && value < min ? `Must be at least ${min}` : undefined
+const minValue18 = minValue(18)
+const email = value =>
+  value && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)
+    ? 'Invalid email address'
+    : undefined
+const tooOld = value =>
+  value && value > 65 ? 'You might be too old for this' : undefined
+const aol = value =>
+  value && /.+@aol\.com/.test(value)
+    ? 'Really? You still use AOL for your email?'
+    : undefined
+const alphaNumeric = value =>
+  value && /[^a-zA-Z0-9 ]/i.test(value)
+    ? 'Only alphanumeric characters'
+    : undefined
+const alphaOnly = value =>
+  value && /[^a-zA-Z ]/i.test(value)
+    ? 'Only alpha characters'
+    : undefined
+
+const renderField = ({ input, label, type, meta: { touched, error, warning } }) => (
```

Comment:
LoginForm and the RegistrationForm could the use same function for rendering of the field as it will ensure that validation is same for both. Now this logic is duplicated.